mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/PathTrackingActivity.java

```diff
@@ -40,19 +48,38 @@ protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_path_tracker);
         ButterKnife.bind(this);
-        start.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                service.startLogging();
-            }
-        });
-
-        stop.setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View view) {
-                service.stopLogging();
-            }
-        });
+
+        if (!CheckSelfPermissionAndRequest
+                .checkSelfPermission(this, Manifest.permission.ACCESS_COARSE_LOCATION)
+                || !CheckSelfPermissionAndRequest
+                .checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION)) {
+            CheckSelfPermissionAndRequest
+                    .requestPermission(this,
+                            Manifest.permission.ACCESS_FINE_LOCATION,
+                            100,
+                            "Location Permission is necessary for this feature",
+                            "Location Permission is necessary for this feature",
+                            "Permission Denied");
+        } else {
+
+            start.setOnClickListener(new View.OnClickListener() {
```

Comment:
`start.setOnClickListener(new View.OnClickListener() {})`

It's there in the master code! I haven't added!

(https://github.com/openMF/android-client/blob/master/mifosng-android/src/main/java/com/mifos/mifosxdroid/activity/PathTrackingActivity.java#L43)